valkyrie/lib/valkyrie/persistence/memory/query_service.rb

```diff
@@ -17,6 +17,14 @@ def find_by(id:)
       cache[id] || raise(::Valkyrie::Persistence::ObjectNotFoundError)
     end
 
+    # @param checksum [Symbol] The checksum to query.
+    # @return [Array<Valkyrie::Resource>] All objects containing the value of the `checksum` property being searched.
+    def find_all_by_checksum(checksum:)
+      cache.values.each.with_object([]) do |obj, result|
```

Comment:
Might be easier to just do `cache.values.select`

Reply:
I tried that at first, but was having a harder time returning an array of `Valkyrie::ID` objects instead of `Valkyrie::Resource` objects. I'll keep digging into it.